akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -758,6 +771,25 @@ private[akka] class ShardRegion(
       context.stop(self) // all shards have been rebalanced, complete graceful shutdown
     }
 
+  def startRegistration(): Unit = {
+    nextRegistrationDelay = initRegistrationDelay
+
+    register()
+    scheduleNextRegistration()
+  }
+
+  def scheduleNextRegistration(): Unit = {
+    if (nextRegistrationDelay < retryInterval) {
+      timers.startSingleTimer(RegisterRetry, RegisterRetry, nextRegistrationDelay)
+      // exponentially increasing retry interval until reaching the normal retryInterval
+      nextRegistrationDelay *= 2
+    }
```

Comment:
Shouldn't it `startSingleTimer` also when reached max delay?

Reply:
nope, then the normal `retryInterval` is used

Reply:
see `timers.startPeriodicTimer(Retry, Retry, retryInterval)` in preStart